### ISSUE TITLE
Intake-API broken

### ISSUE THREAD

**Issue by Karinon** [FREVA-CLINT/databrowserAPI#3]
For some reason I am not allowed to push anything to this Repo, but anyway:

Fabi noticed issues when using the intake api that after ~150 results all result-fields are written as lists (e.g. "project": ["cordex"]).

The issue is in the function `_iterintake` in `core.py`.

Fix could look like that:
```python
                source = {
                    k: out[k][0]
                    if isinstance(out.get(k), list) and len(out.get(k)) == 1
                    else out.get(k)
                    for k in [self.uniq_key] + self.translator.facet_hierachy
                    if out.get(k)
                }
```

More of a generic question:

Is there no better approach how to build a JSON on the fly than putting line breaks manually? The change of functionality is also otherwise visible since the json is no longer properly indented after the mentioned ~150 results.
I haven't done any research on that, so maybe there is nothing better..

**Comment by antarcticrainforest** [comment]
- You should have admin to the repo. So if you want to do the right thing, go ahead and delete it :-P.
- Sorry, I don't understand the problem can you explain a little more - an example would be good.
- Did you forget something between the first if-else clause? To me this isn't valid code.
- I only remember that I had to do this manual linebreak, but don't remember what other alternatives there were. The problem was that json parsers that I found on the python ecosystem or better swamp do not like incomplete json strings/streams which is here the case as it comes from solr.

**Comment by Karinon** [comment]
Ok, probably I was just dumb... now I could create a branch and a pull request

Here is the issue in action:

https://regiklim-dev.dkrz.de/api/databrowser/intake_catalogue/freva/file?translate=false&model=cancm4&start=0

(obviously you need to be in VPN).

The change between "first request" and "subsequent results" starts at line 2378 and looks as follows:

```
      {
         "file": "/work/kd0956/CMIP5/data/cmip5/output1/CCCma/CanCM4/rcp45/mon/atmos/Amon/r6i1p1/v20120612/pr/pr_Amon_CanCM4_rcp45_r6i1p1_200601-203512.nc",
         "project": "cmip5",
         "product": "output1",
         "institute": "CCCma",
         "model": "CanCM4",
         "experiment": "rcp45",
         "time_frequency": "mon",
         "realm": "atmos",
         "variable": "pr",
         "ensemble": "r6i1p1",
         "cmor_table": "Amon",
         "fs_type": "posix",
         "grid_label": "gn"
      },
   {
   "file": "/work/kd0956/CMIP5/data/cmip5/output1/CCCma/CanCM4/rcp45/mon/atmos/Amon/r6i1p1/v20120612/hfls/hfls_Amon_CanCM4_rcp45_r6i1p1_200601-203512.nc",
   "project": [
      "cmip5"
   ],
   "product": [
      "output1"
   ],
   "institute": [
      "CCCma"
   ],
   "model": [
      "CanCM4"
   ],
   "experiment": [
      "rcp45"
   ],
   "time_frequency": [
      "mon"
   ],
   "realm": [
      "atmos"
   ],
   "variable": [
      "hfls"
   ],
   "ensemble": [
      "r6i1p1"
   ],
   "cmor_table": [
      "Amon"
   ],
   "fs_type": "posix",
   "grid_label": [
      "gn"
   ]
},
```

**Comment by antarcticrainforest** [comment]
I had to give admin rights to the freva group. Good to know that this is not done by default. I am closing this now as you've already fixed it in #4 .
